<a href="https://colab.research.google.com/github/gibsonea/Biostats/blob/main/LabSolutions/Solutions04_Overview_of_Probability.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <a name="04intro">2.1: An Overview of Probability</a>

----

Statistical inference is the process of using data from a sample to
describe characteristics of the population. Our conclusions are going to
be based on randomness in the sampling process, and we will need to
account for some uncertainty in our predictions. Statistics relies on
theory and applications from probability to help quantify the
uncertainty in our models and estimates.

-   <font color="mediumseagreen">**If we know the make up of a
    population, we can use probability to predict the likelihood that
    certain outcomes occur.**</font >
    -   If we know a six-sided is fair, what is the probability of
        rolling the die 10 times and getting exactly 4 rolls that land
        on 1?
-   <font color="tomato">**We use statistics to go in the
    reverse direction. Namely, how can we make predictions about a
    population from a random sample of data.**</font >
    -   We want to determine whether or not a six-sided die is fair. We
        roll the die 100 times and get 40 rolls that land on 1. Is the
        die fair?
    -   We will need to use probability to answer statistical questions
        and make inferences about a population.



## <a name="04sample-space">Sample Space, Outcomes and Events</a>

----

A <font color="dodgerblue">**statistical experiment or
observation**</font > is any random activity that results in a definite
outcome.

-   The <font color="dodgerblue">**sample space**</font >, $\color{dodgerblue}{\Omega}$, is the set of all possible outcomes of an experiment.
-   An <font color="dodgerblue">**outcome**</font >, $\color{dodgerblue}{\omega}$, is a result from an experiment or observation.
-   An <font color="dodgerblue">**event**</font >, $\color{dodgerblue}{A}$, is a collection of one or more outcomes from an experiment or observation.



## <a name="04ex-roll">Example: Rolling a Fair Six-Sided Die</a>

----

<figure>
<img
src="https://upload.wikimedia.org/wikipedia/commons/c/c4/2-Dice-Icon.svg"
alt="Image Credit: Adam Spiegler, CC BY-SA 4.0." width = "20%"/>
<figcaption aria-hidden="true">
<a href="https://commons.wikimedia.org/wiki/File:2-Dice-Icon.svg">Steaphan
Greene</a>, <a href="https://creativecommons.org/licenses/by-sa/3.0">CC
BY-SA 3.0</a>, via Wikimedia
Commons
</figcaption>
</figure>


The study of probability was initially inspired by calculating odds of
outcomes from card and dice games. For example, consider rolling a
fair six-sided die.

-   The sample space is $\Omega = \left\{ 1, 2, 3, 4, 5, 6 \right\}$
-   For a fair die, each of the six possible outcomes has an equally likely chance of occurring.
-   One possible outcome is rolling a 4, $\omega = 4$
-   Let $A$ denote the event that the roll is a multiple of 3, $A = \left\{ 3, 6 \right\}$.



## <a name="04equal-likelihood">Probabilities with Equally Likely Outcomes</a>

----

For finite sample spaces, we often use counting to determine
probabilities. A special case which we will encounter often is when each
outcome in the sample space $\Omega$ is equally likely to occur, and
therefore

$$ P(A) = \frac{\mbox{Number of outcomes in event $A$}}{\mbox{Total number of outcomes in $\Omega$}} $$

We use the notation $P(A)$ to denote the probability that event $A$
occurs.

-   Probabilities are proportions between $0$ (impossible to occur) and $1$ (certain to occur) that we typically represent as decimals or fractions.
-   Sometimes we convert the proportion to a percentage when giving a probability.



## <a name="04q1">Question 1</a>

----

If a person rolls a fair, six-sided die, what is the probability the
result of the roll is a number that is divisible by 3?

### <a name="04ans1">Solution to Question 1</a>

----

<br>  


We have

-   The sample space is $\Omega = \left\{ 1, 2, 3, 4, 5, 6 \right\}$
-   Let $A$ denote the event that the roll is a multiple of 3, $A = \left\{ 3, 6 \right\}$.

$$P(A) = \frac{ 2}{ 6 }  = \frac{1}{3} \approx 0.3333 \quad \mbox{or} \quad 33.33\%$$

<br>  

<br>



### <a name="04cardata">Installing and Loading `carData` Package</a>

----

The data set `Arrests` is in the `carData` package in R which is not
installed in Google Colaboratory.

-   First run the code cell below to to install the `carData` package.

In [ ]:
install.packages("carData")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



-   Next load the package with the library command so we can access the
    `Arrests` data set.

In [ ]:
library(carData)

- **We can now access and explore the data set `Arrests` in `carData`.**



## <a name="04q2">Question 2</a>

----

Use R functions such as `summary()`, `str()`, and/or `?Arrests` to
answer some questions about the data:

-   What data is included in the `Arrests` data set?
-   How many observations are in the data?
-   What is the population of interest?
-   What is the source of the data?
-   What are the categorical variables in the data set?
-   What are the quantitative variables in the data set?
-   Are the variable data types accurate, or do some variables need to
    be converted to other data types?



In [ ]:
# Use code cell to summarize and/or get help with Arrests data
?Arrests
str(Arrests)

'data.frame':	5226 obs. of  8 variables:
 $ released: Factor w/ 2 levels "No","Yes": 2 1 2 1 2 2 2 2 2 2 ...
 $ colour  : Factor w/ 2 levels "Black","White": 2 1 2 1 1 1 2 2 1 2 ...
 $ year    : int  2002 1999 2000 2000 1999 1998 1999 1998 2000 2001 ...
 $ age     : int  21 17 24 46 27 16 40 34 23 30 ...
 $ sex     : Factor w/ 2 levels "Female","Male": 2 2 2 2 1 1 2 1 2 2 ...
 $ employed: Factor w/ 2 levels "No","Yes": 2 2 2 2 2 2 1 2 2 2 ...
 $ citizen : Factor w/ 2 levels "No","Yes": 2 2 2 2 2 2 2 2 2 2 ...
 $ checks  : int  3 3 3 1 1 0 0 1 4 3 ...


### <a name="O4ans2">Solution to Question 2</a>

----

Data on police treatment of individuals arrested in Toronto for simple possession of small quantities of marijuana.  There are 5226 observations with 8 variables. The variables are stored correctly.

<br>

The population of interest is all White and Black people arrested for simple possession of small quantities of marijuana in Toronto.

<br>

T
There are 5 categorical variables:

- `released` is whether or not the arrestee was released with a summons. The two levels are `No` and `Yes`.
- `colour` is the arrestee's race. Thre are two levels `Black` and `White`.
- `sex` has levels `Female` and `Male`.
- `employed` has levels `No` and `Yes`.
- `citizen` has levels `No` and `Yes`.

<br>

There are three numerical variables, each stored as an integer:

- `year` is from 1997 through 2002
- `age` in years
- `checks` is the number of police databases on which the arrestee's name appeared.

<br>

The data is from a series of articles in the Toronto Star Newspaper. The data source is listed as "Personal communication from Michael Friendly, York University.".

<br>


## <a name="O4q3">Question 3</a>

----

Suppose you would like to analyze whether female arrestees are more or
less likely to be not released with a summons (and therefore detained in
jail) compared to male arrestees. The two variables of interest are
therefore `sex` and `released`.



### <a name="O4q3a">Question 3a</a>

----

Use the `table()` function, create a two-way table to summarize the
relation between `sex` and `released`.

#### <a name="O4ans3a">Solution to Question 3a</a>

----

In [ ]:
summary(Arrests)

 released     colour          year           age            sex      
 No : 892   Black:1288   Min.   :1997   Min.   :12.00   Female: 443  
 Yes:4334   White:3938   1st Qu.:1998   1st Qu.:18.00   Male  :4783  
                         Median :2000   Median :21.00                
                         Mean   :2000   Mean   :23.85                
                         3rd Qu.:2001   3rd Qu.:27.00                
                         Max.   :2002   Max.   :66.00                
 employed   citizen        checks     
 No :1115   No : 771   Min.   :0.000  
 Yes:4111   Yes:4455   1st Qu.:0.000  
                       Median :1.000  
                       Mean   :1.636  
                       3rd Qu.:3.000  
                       Max.   :6.000  

In [ ]:
table(Arrests$sex,Arrests$released)

        
           No  Yes
  Female   63  380
  Male    829 3954

### <a name="O4q3b">Question 3b</a>

----

You should see from your output in Question 3a that more Male arrestees
were not released (829) compared the female arrestees that were not released
(63). Why is it problematic to compare counts of female and male
arrestees and conclude female arrestees are more likely to be detained
(not be released)?

#### <a name="O4ans3b">Solution to Question 3b</a>

----


There are about ten times as many male arrestees (4783) in the sample compared to female arrestees (443). If being released is independent of gender, we'd expected more male arrestees in both the released and not released groups just because we have more male observations in the data.
<br>



### <a name="O4q3c">Question 3c</a>

----

What is the probability that a randomly selected arrestee in the study:

1.  Was not released?
2.  Was male?
3.  Was not released and was male?
4.  Was not released or was male?
5.  Given that a person was male, what is the probability they were not
    released?
6. Given that a person was female, what is the probability they were not released?

#### <a name="O4ans3c">Solution to Question 3c</a>

----

Use the code cell below to compute each of the probabilities (1)-(5).

In [ ]:
# Enter code to compute each of the following
# Be sure to print results to screen!
# Find the probability that a randomly selected person in the study:

total <- nrow(Arrests)  # count number or rows (observations) in Arrests
total

# (1) was not released

notrel <- Arrests$released == "No"
P.notrel = sum(notrel)/total
P.notrel

# (2) was Male

male <- Arrests$sex == "Male"
P.male = sum(male)/total
P.male

# (3) was not released and was Male
my.tbl = table(Arrests$released,Arrests$sex)
my.tbl
my.tbl[1,2]/total

# (4) was not released or was Male
(sum(notrel)+sum(male)-my.tbl[1,2])/total

# (5) was not released given they were Male
my.tbl[1,2]/sum(male)


[1] 5226

[1] 0.170685

[1] 0.9152315

     
      Female Male
  No      63  829
  Yes    380 3954

[1] 0.1586299

[1] 0.9272866

[1] 0.1733222

Summarize results below

1.  The probability that a randomly selected person in the study was not released is $\color{dodgerblue}{P(N)=0.171}$.

2.  The probability that a randomly selected person in the study was Male is $\color{dodgerblue}{P(M)=0.915}$.

3.  The probability that a randomly selected person in the study was not released and was Male is $\color{dodgerblue}{P(N \cap M)=0.159}.$

4.  The probability that a randomly selected person in the study was not released or was Male is $\color{dodgerblue}{P(N \cup M)=0.927}$.

5.  Given that a person was Male, what is the probability they were not released?

$$\color{dodgerblue}{P(N \ | \ M)=0.173}$$



### <a name="O4q3d">Question 3d</a>

----

Based on the data from this study, do you believe Male arrestees are
more, less, or equally likely to be detained (not be released) than
Female arrestees? Support your answer using probabilities. You may want
to compute additional probabilities that were not asked in [Question
3c](#04q3c) before reaching your conclusion.

#### <a name="O4ans3d">Solution to Question 3d</a>

----

One way to analyze this question is to compare the number female arrestees that were not released relative to size of the female sample size, $P( N \ | \ F)$, to the number of male arrestees that were not released relative to size of the male sample size $P( N \ | \ M)$

Using the frequency two way table above, we have

- $P(N \ | \ F)= \frac{63}{443} \approx 0.142$
- $P(N \ | \ M)= \frac{829}{4783} \approx 0.173$

<font color="dodgerblue">Comparing the two conditional probabilities above, we see male arrests are more likely to not be released than female arrestees.</font>

The table stored in `my.tbl` has each gender as a different column, so if we construct a `prop.table` that gives the proportions relative to the columns (using option `2`), we can check our previous answers.
<br>



In [ ]:
#additional calculations

# was not released given that they are female

female <- Arrests$sex == "Female"
my.tbl[1,1]/sum(female)

prop.table(my.tbl,2)

[1] 0.1422122

     
         Female      Male
  No  0.1422122 0.1733222
  Yes 0.8577878 0.8266778

# <a name="O4notation">Simple and Conditional Probabilities</a>

----

Let $A$ and $B$ denote two events in sample space $\Omega$, then

-   $\color{dodgerblue}{P(A)}$ is the probability that event $A$ occurs.
-   $\color{dodgerblue}{P(\bar{A}) }$ is the probability that event $A$ <font color="dodgerblue">**does NOT occur**</font >.
-   The notation $\color{dodgerblue}{\bar{A}}$, is used to denote the <font color="dodgerblue">**complement**</font > of $A$.
-   $\color{dodgerblue}{P(A \cap B)}$ is the probability that events $A$ <font color="dodgerblue">**and**</font > $B$ both occur.
-   $\color{dodgerblue}{P(A \cap \bar{B})}$ is the probability that <font color="dodgerblue">**event A occurs and event B does not occur**</font >.
-   $\color{dodgerblue}{P(A \cup B)}$ is the probability that either event $A$ <font color="dodgerblue">**or**</font > event $B$ occurs (or both $A$ and $B$ occur).
-   $\color{dodgerblue}{P(B \ | \ A )}$ is the <font color="dodgerblue">**conditional probability**</font > that event $B$ occurs <font color="dodgerblue">**given that**</font > event $A$ occurs.




# <a name="04prop">Properties of Probability</a>

----

We can generalize the calculations from the previous study on people
arrested for small quantities of marijuana to obtain the following
results:

Let $A$ and $B$ denote two events in sample space $\Omega$, then

-   <font color="dodgerblue">**Additive property**</font >:
    $P(A \cup B) = P(A) + P(B) - P(A \cap B)$.
-   <font color="dodgerblue">**Conditional Probability**</font >:
    $\displaystyle P(B | A) = \frac{P(A \cap B)}{P(A)}$
-   <font color="dodgerblue">**Multiplicative property**</font >:
    $P(A \cap B) = P(A) \cdot P(B | A)$
-   <font color="dodgerblue">**Complement property**</font >:
    $P(\bar{A}) = 1 - P(A)$



## <a name="04q4">Question 4</a>

----

Match one of the <font color="dodgerblue">**Venn
diagrams**</font > labelled (i)-(vi) in the table below to one of the set
operations below. Note that two of the Venn Diagrams do not match any of
the set operations.

| <font size=3>diagram (i)</font>  | <font size=3>diagram (ii)</font>  | <font size=3>diagram (iii)</font> |
|--------------|---------------|----------------|
| <img src="https://upload.wikimedia.org/wikipedia/commons/6/60/04fig-venn-001.png" alt="Image Credit: Adam Spiegler, CC BY-SA 4.0."  /> | <img src="https://upload.wikimedia.org/wikipedia/commons/0/0f/04fig-venn-002.png" alt="Image Credit: Adam Spiegler, CC BY-SA 4.0." /> | <img src="https://upload.wikimedia.org/wikipedia/commons/c/c4/04fig-venn-003.png" alt="Image Credit: Adam Spiegler, CC BY-SA 4.0."  />  |


| <font size=3>diagram (iv)</font>  | <font size=3>diagram (v)</font> | <font size=3>diagram (vi)</font>   |
|---------------|-------------|----------------|
| <img src="https://upload.wikimedia.org/wikipedia/commons/7/73/04fig-venn-004.png" alt="Image Credit: Adam Spiegler, CC BY-SA 4.0."  /> | <img src="https://upload.wikimedia.org/wikipedia/commons/7/77/04fig-venn-008.png" alt="Image Credit: Adam Spiegler, CC BY-SA 4.0."  />| <img src="https://upload.wikimedia.org/wikipedia/commons/9/9e/04fig-venn-005.png" alt="Image Credit: Adam Spiegler, CC BY-SA 4.0."  /> |

Image credit: Adam Spiegler, [CC BY-SA
4.0](https://creativecommons.org/licenses/by-sa/4.0), via Wikimedia
Commons

### <a name="04ans4">Solution to Question 4</a>

----

| $\large{\bar{A}}$      | $\large{A \cup B}$ | $\large{A \cap B}$ | $\large{A \cap \bar{B}}$   |
|------------|------------|------------|------------|
| <font size=3>Diagram (vi)</font> | <font size=3>Diagram (ii)</font> | <font size=3>Diagram (i)</font> | <font size=3>Diagram (iii)</font> |


<br> <br> <br>



# <a name="04ind">Independent Events</a>

----

Often in statistics we want to investigate questions such as:

-   Is a newly developed vaccine effective?
-   Do certain sentencing laws have an effect on crime rates?
-   Did increasing the minimum wage for fast food workers effect fast food prices?
-   **Does the occurrence of one event** $M$ (being male) **effect the likelihood that another event** $N$ (being detained after arrest) **occurs**?



## <a name="04def-ind">Definition of Independent Events</a>

----

Two events $A$ and $B$ are
<font color="dodgerblue">**independent**</font > if the
occurrence of one has no effect on the occurrence of the other:

$$ P(B) = P(B \ | \ A) \quad \mbox{or} \quad P(A) = P(A \ | \ B).$$

**Special case:** If events $A$ and $B$ are independent events then we
have $P(A \cap B) = P(A)P(B)$.



## <a name="04q5">Question 5</a>

----

Suppose the probability that CU football will beat CSU is 70%, while the probability that CU will beat Kansas is 40%.  

$$ P(A) = 0.7, P(B) = 0.4 $$

Where the event A = {CU beats CSU} and event B = {CU beats KU}.




### <a name="04q5a">Question 5a</a>

----

What is the probability that CU football wins both games?  How would you write this in probability notation?

#### <a name="04ans5a">Solution to Question 5a</a>

----

Assume that each event is independent.  We want to calculate the probability of A **and** B or the overlap:
$P(A \cap B) = P(A) \cdot P(B) = 0.7 \cdot 0.4 = 0.28$  or $28 \%$
<br>



### <a name="04q5b">Question 5b</a>

----

What is the probability that CU wins **at least** one game?  How would you write this in probability notation?

#### <a name="04ans5b">Solution to Question 5b</a>

----

We want to calculate the probability of A **or** B which is expressed as $P(A \cup B)$.  Remember you have to sum the probabilities but then subtract the overlap so it isn't counted twice.
<br>  
$P(A \cup B) = P(A) + P(B) - P(A \cap B) = 0.7 + 0.4 - (0.7 \cdot 0.4) =$
$ 1.1 - 0.28 = 0.82 $ or $ 82\% $
<br>



### <a name="04q5c">Question 5c</a>

----

What is the probability that CU wins only one game?  Again, write in probability notation.

#### <a name="04ans5c">Solution to Question 5c</a>

----

We want to sum the probability of A and probability of B but subtract the overlap:
<br>

$P(A \cup B) - P(A \cap B) = [P(A) + P(B) - P(A \cap B)] - P(A \cap B) = $
$[0.7 + 0.4 - 0.7 \cdot 0.4] - 0.7 \cdot 0.4 = 0.82 - 0.28 = 0.54$ or $54 \% $
<br>
<br>



### <a name="04ans5d">Question 5d</a>

----

What is the probability that CU wins no games?  Again, write in probability notation.

#### <a name="04q5d">Solution to Question 5d</a>

----

Option #1: Probability of event A not occuring **and** event B not occuring: $ P(\bar{A} \cap \bar{B})$.  Using the property of complements:
$$ P( \bar{A}) = 1 - P(A) = 1 - 0.7 = 0.3$$
$$ P(\bar{B}) = 1 - P(B) = 1 - 0.4 = 0.6$$


$$ P(\bar{A} \cap \bar{B}) = Pr(\bar{A}) \cdot Pr(\bar{B}) = 0.3 \cdot 0.6 = 0.18$$
<br>
Option #2: The probability of CU winning no games is also the same as the complement of the probability of CU winning at least one game:
$$ 1 - P(A \cup B) = 1 - 0.82 = 0.18 $$


## <a name="04q6">Question 6</a>

----

Consider the board game picture below. You begin at the start position
and flip a fair coin. If the result of the flip is tails, you move
forward one space on the board. If the result is heads, you move forward
two spaces on the board. Keep flipping a fair coin (with tails moving
one space and heads two spaces) until you reach the winner position.

<figure>
<img
src="https://upload.wikimedia.org/wikipedia/commons/d/d9/Simple-boardgame.png"
alt="Image Credit: Adam Spiegler, CC BY-SA 4.0." />
<figcaption aria-hidden="true">
Image Credit: Adam Spiegler, CC BY-SA 4.0
<https://creativecommons.org/licenses/by-sa/4.0>, via Wikimedia Commons
</figcaption>
</figure>

-   What is the probability that you win in each exactly two flips?
-   What is the probability that you win in each exactly three flips?
-   What is the probability that you win in each exactly four flips?

### <a name="04ans6">Solution to Question 6</a>

----

In order to win in exactly two flips, that requires first rolling a tail and then rolling a head.

$$P( \mbox{win in exactly two flips} ) = P(TH) = P(T) P(H) = 0.25.$$

In order to win in exactly three flips, a person could roll $HTH$: a head, then a tail, and then a head.

$$P( \mbox{win in exactly three flips} ) = P(HTH) = P(H) P(T) P(H) = 0.125.$$

In order to win in exactly four flips, a person could roll:

- $HHTH$ - Heads, Heads, Tails, then Heads.
- $TTTH$ - Tails, Tails, Tails, then Heads.

$$P (\mbox{win in exactly four flips} ) = P(HHTH) + P(TTTH) = 0.125.$$
<br>



## <a name="04q7">Question 7</a>

----

Does having health insurance help avoid bankruptcies? Let $B$ denote the
event a person goes bankrupt. Let $H$ denote the event a person has
health insurance.



### <a name="04q7a">Question 7a</a>

----

What is the difference in the practical meaning $P(H \ | \ B)$ and
$P(B \ | \  H )$? Explain in practical terms that a non-statistician
could understand, and avoiding technical language.

#### <a name="04ans7a">Solution to Question 7a</a>

----


- $P(H \ | \ B)$ is the probability that a person has health insurance given they have gone bankrupt.

- $P(B\ | \ H)$ is the probability that a person has gone bankrupt given they have health insurance.
<br>



### <a name="04q7b">Question 7b</a>

----

Can you determine whether having health insurance has any affect on the
likelihood that a person goes bankrupt by comparing $P(H \ | \ B)$ and
$P(B \ | \ H )$?

-   If so, explain how you would compare those to probabilities to help
    answer the question.
-   If not, what additional probability (or probabilities) would be
    useful to know in order to answer this question.

#### <a name="04ans7b">Solution to Question 7b</a>

----


No, it is not very useful to only compare $P(H \ | \ B)$ with $P(B\ | \ H)$ as they are measuring different probabilities that are not relevant to compare. If we want to know whether $H$ and $B$ are independent, then we could compare $P(H \ | \ B)$ with $P(H)$ and/or $P(H \ | \ \bar{B})$. Otherwise, we could choose to compare $P(B \ | \ H)$ with $P(B)$ and/or $P(B \ | \ \bar{H})$ instead.
<br>



# <a name="04excl">Mutually Exclusive Events</a>

----

Two events $A$ and $B$ are <font color="dodgerblue">**mutually
exclusive**</font > (or
<font color="dodgerblue">**disjoint**</font >) if they cannot
occur at the same time, and therefore $P(A \cap B) = 0$.

**Special case:** If events $A$ and $B$ are disjoint then we have
$P(A \cup B) = P(A)+P(B)$.



## <a name="04q8">Question 8</a>

----

Give an example of two events that are mutually exclusive, and give an
example of two events that are not mutually exclusive.

#### <a name="04ans8">Solution to Question 8</a>

----

Many possible answers to give. Below is one possible answer (for each).

- Mutually Exclusive: The events "a person is born in May" and "a person is born in June" would be mutually exclusive events. A person cannot be born in two different months.

- Not Mutually Exclusive: The events "a person is born in May" and "a person has brown eyes". A person can both have brown eyes and be born in May.
<br>



# <a name="04prob-dist">Probability Distributions</a>

----

A function $P$ that assigns a real number $P(A)$ to each event $A$ is a
<font color="dodgerblue">**probability distribution**</font > or
a <font color="dodgerblue">**probability measure**</font > if it
satisfies the following three axioms:

-   $0 \leq P(A) \leq 1$ for all $A$.
-   $P(\mbox{full sample space})=P(\Omega) = 1$.
-   If $A$ and $B$ are disjoint, then

$$P \left( A \cup B \right) = P(A) + P(B).$$



# <a name="CC License">Creative Commons License Information</a>
---


![Creative Commons
License](https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png)

*Statistical Methods: Exploring the Uncertain* by [Adam
Spiegler (University of Colorado Denver)](https://github.com/CU-Denver-MathStats-OER/Statistical-Theory)
is licensed under a [Creative Commons
Attribution-NonCommercial-ShareAlike 4.0 International
License](http://creativecommons.org/licenses/by-nc-sa/4.0/). This work is funded by an [Institutional OER Grant from the Colorado Department of Higher Education (CDHE)](https://cdhe.colorado.gov/educators/administration/institutional-groups/open-educational-resources-in-colorado).

For similar interactive OER materials in other courses funded by this project in the Department of Mathematical and Statistical Sciences at the University of Colorado Denver, visit <https://github.com/CU-Denver-MathStats-OER>.